core/mm/core_mmu.c

```diff
@@ -2060,11 +2060,16 @@ void *core_mmu_add_mapping(enum teecore_memtypes type, paddr_t addr, size_t len)
 }
 
 #ifdef CFG_WITH_PAGER
-static vaddr_t get_linear_map_end(void)
+static paddr_t get_linear_map_end_va(void)
 {
 	/* this is synced with the generic linker file kern.ld.S */
 	return (vaddr_t)__heap2_end;
 }
+
+static paddr_t get_linear_map_end_pa(void)
+{
+	return get_linear_map_end_va() - VCORE_START_VA + TEE_LOAD_ADDR;
```

Comment:
cast?
`return  (paddr_t)(get_linear_map_end_va() - VCORE_START_VA + TEE_LOAD_ADDR);`

Reply:
I'm sorry, I can't see the point with that.

Reply:
do we need here to explicitly cast from `vaddr_t` to `paddr_t`?

Reply:
I'm not sure what would be gained by that. The type is already converted automatically.

Reply:
ok, fine.